packages/core/http/core-http-router-server-internal/src/versioned_router/core_versioned_route.ts

```diff
@@ -109,6 +108,13 @@ export class CoreVersionedRoute implements VersionedRoute {
         body: `No handlers registered for [${this.method}] [${this.path}].`,
       });
     }
+
+    if (!this.hasVersion(req) && (this.isInternal || this.router.isDev)) {
```

Comment:
The title says "for internal and when in dev mode", which I initially took to mean AND, but the code implies OR - is it the latter? just to confirm

Reply:
The intention is require versions:

(1) Always for internal endpoints
(2) Only during dev for public endpoints

 I'll try to clarify the title.